File: Assignment_1_1.py
a = [1,2,3,4,5]
x = 4
prev = a[0]
next = a[-1]
while prev<=next:
    s = prev + next 
    if (s==x):#x
        break
    elif (s<x):
        prev += 1
    else:
        next -= 1
print(prev,next)

'''
#with for loop
for i in range(len(a)):
    if prev>=next:
        break
    s = prev + next
    if (s==x):
        print(prev,next)
        break
    elif(s<x):
        a = a[1:] #remove first element
    else:
        a = a[:-1] #remove last element
'''

File: Assignment_1_1.c
//#given a sorted element and num x ,search two elements and sum should be x

#include <stdio.h>

int main() {
    int a[] = {1, 2, 3, 4, 5};
    int x = 4;
    int prev = a[0];
    int next = a[4];  // Last element (a[-1] in Python)
    
    while (prev <= next) {
        int s = prev + next;
        if (s == x) {  // x
            break;
        } else if (s < x) {
            prev++;
        } else {
            next--;
        }
    }
    
    printf("%d %d\n", prev, next);
    return 0;
}